pkg/scheduler/objects/reservation.go

```diff
@@ -21,37 +21,36 @@ package objects
 import (
 	"go.uber.org/zap"
 
-	"github.com/apache/yunikorn-core/pkg/common"
 	"github.com/apache/yunikorn-core/pkg/log"
 )
 
 type reservation struct {
-	nodeID string
-	appID  string
-	askKey string
-	// these references must ONLY be used for ask, node and application removal otherwise
+	appID    string
+	nodeID   string
+	allocKey string
+	// these references must ONLY be used for alloc, node and application removal otherwise
 	// the reservations cannot be removed and scheduling might be impacted.
-	app  *Application
-	node *Node
-	ask  *Allocation
+	app   *Application
+	node  *Node
+	alloc *Allocation
 }
 
 // The reservation inside the scheduler. A reservation object is never mutated and does not use locking.
 // The key depends on where the reservation was made (node or app).
 // appBased must be true for a reservation for an app and false for a reservation on a node
-func newReservation(node *Node, app *Application, ask *Allocation, appBased bool) *reservation {
-	if ask == nil || app == nil || node == nil {
+func newReservation(node *Node, app *Application, alloc *Allocation, appBased bool) *reservation {
```

Comment:
I know it's a matter of semantics, but can we call this allocation at this point? It's funny because now we have the same object for both an ask and alloc... This occurs multiple places in this patch. Don't want to nitpick on this one, just wondering which naming convention is better.

Reply:
My personal view on this: Use whatever name makes sense given the context. If it's known we're dealing with an unsatisified allocation, ask seems fine. It helps give some context. In the case of reservations, I think ask is correct, since once we satisfy an allocation, it shouldn't interact further with reservations.

Reply:
I don't want to be pedantic either way though.